### PR TITLE
ON-241

### PR DESCRIPTION
**Pull request by Dvi55**

### CODE REVIEW

File: src/main/java/com/gameshop/www/eCommerce/cart/controller/CartController.java [hunk 1]
```diff
@@ -1,47 +1,47 @@
...
 import org.springframework.http.ResponseEntity;
 import org.springframework.security.core.annotation.AuthenticationPrincipal;
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.RequestBody;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
 
 import java.util.List;
 
 @AllArgsConstructor
```
Comment (SerhiiVashkulat) [inline src/main/java/com/gameshop/www/eCommerce/cart/controller/CartController.java]: @RequiredArgsConstructor better

File: src/main/java/com/gameshop/www/eCommerce/cart/controller/CartController.java [hunk 2]
```diff
@@ -1,47 +1,47 @@
...
-@RestController("/cart")
+@RestController
+@RequestMapping("/cart")
```
Comment (SerhiiVashkulat) [inline src/main/java/com/gameshop/www/eCommerce/cart/controller/CartController.java]: REST  "/carts"

File: src/main/java/com/gameshop/www/eCommerce/cart/controller/CartController.java [hunk 3]
```diff
@@ -1,47 +1,47 @@
...
-    @PostMapping("/details")
-    public ResponseEntity<Cart> getCartDetails(@AuthenticationPrincipal LocalUser user) {
-        Cart cart = cartService.getCartDetails(user);
+    @GetMapping("/details")
+    public ResponseEntity<CartDto> getCartDetails(@AuthenticationPrincipal LocalUser user) {
+        CartDto cart = cartService.getCartDetails(user);
         return ResponseEntity.ok(cart);
     }
 
     @PostMapping("/remove")
```
Comment (SerhiiVashkulat) [inline src/main/java/com/gameshop/www/eCommerce/cart/controller/CartController.java]: @DeleteMapping

File: src/main/java/com/gameshop/www/eCommerce/cart/model/CartItem.java [hunk 4]
```diff
@@ -22,14 +23,15 @@ public class CartItem {
...
     private Product product;
 
     @Column(name = "quantity")
     private Integer quantity;
 
-    @ManyToOne(cascade = CascadeType.ALL)
-    @JoinColumn(name = "cart_id")
+    @JsonBackReference
+    @ManyToOne
+    @JoinColumn
```
Comment (SerhiiVashkulat) [inline src/main/java/com/gameshop/www/eCommerce/cart/model/CartItem.java]: @JoinColumn(name = "cart_id") I think  need reference column to relation and product

File: src/main/java/com/gameshop/www/eCommerce/cart/service/CartService.java [hunk 5]
```diff
@@ -2,28 +2,28 @@
...
+import lombok.AllArgsConstructor;
 import org.springframework.stereotype.Service;
 
-import java.util.List;
-import java.util.Map;
-import java.util.UUID;
+import java.util.*;
 import java.util.stream.Collectors;
-
+@AllArgsConstructor
```
Comment (SerhiiVashkulat) [inline src/main/java/com/gameshop/www/eCommerce/cart/service/CartService.java]: My mistake ask you different annotation sorry, use @requiredargsconstructor

File: src/main/java/com/gameshop/www/eCommerce/cart/service/CartService.java [hunk 6]
```diff
@@ -34,54 +34,42 @@ public Cart getCartByUser(LocalUser user) {
...
 
     @Transactional
-    public Cart addProductToCart(LocalUser user, List<CartDto> cartDtos) {
-        Cart cart = getCartByUser(user);
-
-        List<UUID> productIds = cartDtos.stream()
-                .map(CartDto::getId)
-                .collect(Collectors.toList());
+    public CartDto addProductToCart(LocalUser user, List<CartBody> cartBodies) {
+        LocalUser persistedUser = localUserDAO.findById(user.getId()).orElseThrow(() -> new RuntimeException("User not found"));
```
Comment (SerhiiVashkulat) [inline src/main/java/com/gameshop/www/eCommerce/cart/service/CartService.java]: UserNotFoundException

File: src/main/java/com/gameshop/www/eCommerce/cart/service/CartService.java [hunk 7]
```diff
@@ -34,54 +34,42 @@ public Cart getCartByUser(LocalUser user) {
...
+        Cart cart = getCartByUser(persistedUser);
```
Comment (SerhiiVashkulat) [inline src/main/java/com/gameshop/www/eCommerce/cart/service/CartService.java]: Mb create 1 private method and use it in the publics other methods. findCartByUserId

**Review by Liyto1 — APPROVED** [review]

**Pull request merged**